### PR TITLE
[SMALLFIX] Configuration parameter for UFS listing length

### PR DESCRIPTION
**Pull request by madanadit**
Large directory list and delete test pagination in the UFS. To speed up testing the parameter "alluxio.underfs.listing.length" can be set to a small number, allowing testing pagination for smaller directories.

S3 returns a maximum of 1000 list items. Didn't find similar documentation for other UFSs.

### CODE REVIEW

File: tests/pom.xml [hunk 1]
```diff
@@ -337,6 +339,7 @@
                 <!-- Change "container/folder" to contain the testing container  -->
                 <!-- Keep the structure in form container/folder -->
                 <containerKey>swift://container/folder</containerKey>
+                <alluxio.underfs.listing.length>50</alluxio.underfs.listing.length>
```
Comment (calvinjia) [inline tests/pom.xml]: Would it be possible to set this config value in the test instead of the profile?

**Comment by ifcharming** [comment]
LGTM

**Comment by madanadit** [comment]
@calvinjia Please take another look.

File: tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java [hunk 2]
```diff
@@ -45,6 +45,7 @@
 
   @Before
   public final void before() throws Exception {
+    Configuration.set(PropertyKey.UNDERFS_LISTING_LENGTH, 50);
```
Comment (calvinjia) [inline tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java]: Could you reset the configuration in an `After`?

File: tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java [hunk 3]
```diff
@@ -42,13 +43,25 @@
...
     mUnderfsAddress = Configuration.get(PropertyKey.UNDERFS_ADDRESS);
     mUfs = UnderFileSystem.get(mUnderfsAddress + AlluxioURI.SEPARATOR);
   }
 
+  @After
+  public final void after() throws Exception {
+    if (mUfsListingLength > 0) {
+      Configuration.set(PropertyKey.UNDERFS_LISTING_LENGTH, mUfsListingLength);
+    }
+  }
```
Comment (calvinjia) [inline tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java]: would `ConfigurationTestUtils.resetConfiguration()` work?

Comment (madanadit) [inline tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java]: Thanks!. I did not know there was this utility function.

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 4]
```diff
@@ -369,6 +370,7 @@
     // UFS related properties
     //
     public static final String UNDERFS_ADDRESS = "alluxio.underfs.address";
+    public static final String UNDERFS_LISTING_LENGTH = "alluxio.underfs.listing.length";
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/PropertyKey.java]: Could you add docs for this new configuration value?

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**